source/ceylon/time/iso8601/parseTimeZone.ceylon

```diff
@@ -1,200 +1,67 @@
-import ceylon.time.base { ms = milliseconds }
 import ceylon.time.timezone {
-    timeZone,
-    TimeZone
+    TimeZone,
+    OffsetTimeZone,
+    timeZone
+}
+import ceylon.time.base {
+    milliseconds
 }
-
-
 
 "Timezone offset parser based on ISO-8601, currently it accepts the following 
  time zone offset patterns:
  
- - &plusmn;`[hh]:[mm]`, 
+ - &plusmn;`[hh]:[mm]`,
  - &plusmn;`[hh][mm]`, and 
  - &plusmn;`[hh]`.
  
  In addition, the special code `Z` is recognized as a shorthand for `+00:00`"
-shared TimeZone|ParserError parseTimeZone( String offset ) {
-    variable TzState state = TzInitial();
-    for( character in offset ) {
-        state = state.next( Chunk( character ) );
+shared TimeZone? parseTimeZone( String offset ) {
+    if(offset.equalsIgnoringCase("Z")) {
```

Comment:
Off the top of my head, but I don't think the lower case "z" should be allowed here...